src/main/java/Duke.java

```diff
@@ -1,10 +1,69 @@
+import java.util.Scanner;
+import java.util.Arrays;
+
 public class Duke {
+    public static void printLine() {
+        System.out.println("______________________________________________________________");
+    }
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String line;
+        String lowerCaseLine;
+        Task[] textList = new Task[100]; //store in array of Tasks to track both description and mark status
+        int textListCount = 0;
+        Scanner in = new Scanner(System.in);
+
+        printLine();
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        printLine();
+
+        while(true) {
+            line = in.nextLine();
+            lowerCaseLine = line.toLowerCase();
+            Task task = new Task(line);
+            printLine();
+            if (lowerCaseLine.equals("bye")) {
+                System.out.println("Bye. Hope to see you again soon!");
+                printLine();
+                return;
+            } else if (lowerCaseLine.equals("list")) { //if input is list
+                System.out.println("Here are the tasks in your list: ");
+                for (int i = 0; i < textListCount; ++i) {
+                    String listLine = i+1 + ". " + textList[i].toString();
+                    System.out.println(listLine);
+                }
+                printLine();
+            } else if (lowerCaseLine.contains("mark") && !lowerCaseLine.contains("unmark")) { //mark is in unmark so need to check like this to avoid unmark being detected as mark
+                int indexOfTask = Integer.parseInt(lowerCaseLine.substring(5)) - 1; //find index of the task number starting from 0
+                textList[indexOfTask].markAsDone();
+                System.out.println("Nice! I've marked this task as done: ");
+                System.out.println(" " + textList[indexOfTask].toString());
+                printLine();
+            } else if (lowerCaseLine.contains("unmark")) {
+                int indexOfTask = Integer.parseInt(lowerCaseLine.substring(7)) - 1;
+                textList[indexOfTask].unmarkAsDone();
+                System.out.println("OK, I've marked this task as not done yet: ");
+                System.out.println(" " + textList[indexOfTask].toString());
+                printLine();
+            }
+            else { //if input is any other text
```

Comment:
Nested if-else statements in else statement. Perhaps you can extract this out and separate them?